### PR TITLE
Fixes a bug where, in sac.py, self._alpha was not being re-computed a…

### PR DESCRIPTION
**Pull request by jordan-schneider**
…fter loading self._log_alpha from an optim_state_dict.

### CODE REVIEW

**Comment by jordan-schneider** [comment]
Alternatively, line 112 `self._alpha = torch.exp(self._log_alpha.detach())` could be moved after the `self.load_optim_state_dict()` call, which is slightly faster, but it logically separates the self._alpha definition from both the self._log_alpha definitions, which I feel is much less readable.

**Comment by astooke** [comment]
Oops, good catch.  Yes I also prefer the way you did it, thanks!
